### ISSUE TITLE
Document running your own server

### ISSUE THREAD

**Issue by 0dragosh** [ellie/atuin#197]
Currently, the documentation on how to run your own server is lacking completely. 

It would be nice to have some docs around that especially since the help output of `atuin server start --help` isn't very, well, helpful :)

**Comment by aaronnewsome** [comment]
A few days ago I setup an atuin server for myself. It was only by accident that I figured out that I could use a postgres server instead of sqlite. I didn't really see any documentation on setting up the postgres server. I didn't know if there was some schema DDL I needed to import but luckily I just created the DB, created the config file, started the atuin server and it created all the tables, etc.

There is a Dockerfile in this github, but it didn't work for me. After building the docker container from the Dockerfile in this github, the atuin binary gives an error about wrong GLIBC version. So I created my own Dockerfile for building the atuin server. Here's the complete Dockerfile:

```
FROM ubuntu:20.04
ENV DEBIAN_FRONTEND  noninteractive
WORKDIR /
RUN ln -fs /usr/share/zoneinfo/America/Los_Angeles /etc/localtime
RUN apt-get update && apt-get -y install build-essential autoconf automake gdb git libffi-dev zlib1g-dev libssl-dev curl && rm -rf /var/lib/apt/lists/*
RUN echo 'root:$1$tWm0tdKj$UvyLFsC9l5h3w436eWtY0' | chpasswd -e
RUN curl https://sh.rustup.rs -sSf | sh -s -- -y 
RUN . $HOME/.cargo/env
RUN echo ". $HOME/.cargo/env" >> $HOME/.bashrc
RUN ["/bin/bash", "-c", "source $HOME/.cargo/env ; rustup install stable"]
RUN ["/bin/bash", "-c", "source $HOME/.cargo/env ; rustup default stable"]
RUN ["/bin/bash", "-c", "source $HOME/.cargo/env ; cargo install atuin"]
WORKDIR /root
RUN mkdir -pv $HOME/.config/atuin/
RUN echo "host=\"0.0.0.0\"" > $HOME/.config/atuin/server.toml
RUN echo "port=8000"              >> $HOME/.config/atuin/server.toml
RUN echo "open_registration=true" >> $HOME/.config/atuin/server.toml
RUN echo "db_uri=\"postgres://atuin:dbPassword@postgres.homelab.local/atuin\"" >>  $HOME/.config/atuin/server.toml
RUN echo "#!/bin/bash" > $HOME/startServer
RUN echo "source $HOME/.bashrc" >> $HOME/startServer
RUN echo "source $HOME/.cargo/env" >> $HOME/startServer
RUN echo "atuin server start" >> $HOME/startServer
RUN chmod -R og-rwx $HOME/.config
RUN chmod +x $HOME/startServer
ENV ATUIN_CONFIG_DIR /root/.config/atuin
ENV ATUIN_PORT 8000
CMD $HOME/startServer
```

Even if you're not using docker, you can probably see how the atuin server is configured and started. 

It's not documentation, but hope it helps.

**Comment by 0dragosh** [comment]
> A few days ago I setup an atuin server for myself. It was only by accident that I figured out that I could use a postgres server instead of sqlite. I didn't really see any documentation on setting up the postgres server. I didn't know if there was some schema DDL I needed to import but luckily I just created the DB, created the config file, started the atuin server and it created all the tables, etc.
> 
> There is a Dockerfile in this github, but it didn't work for me. After building the docker container from the Dockerfile in this github, the atuin binary gives an error about wrong GLIBC version. So I created my own Dockerfile for building the atuin server. Here's the complete Dockerfile:
> 
> ```
> FROM ubuntu:20.04
> ENV DEBIAN_FRONTEND  noninteractive
> WORKDIR /
> RUN ln -fs /usr/share/zoneinfo/America/Los_Angeles /etc/localtime
> RUN apt-get update && apt-get -y install build-essential autoconf automake gdb git libffi-dev zlib1g-dev libssl-dev curl && rm -rf /var/lib/apt/lists/*
> RUN echo 'root:$1$tWm0tdKj$UvyLFsC9l5h3w436eWtY0' | chpasswd -e
> RUN curl https://sh.rustup.rs -sSf | sh -s -- -y 
> RUN . $HOME/.cargo/env
> RUN echo ". $HOME/.cargo/env" >> $HOME/.bashrc
> RUN ["/bin/bash", "-c", "source $HOME/.cargo/env ; rustup install stable"]
> RUN ["/bin/bash", "-c", "source $HOME/.cargo/env ; rustup default stable"]
> RUN ["/bin/bash", "-c", "source $HOME/.cargo/env ; cargo install atuin"]
> WORKDIR /root
> RUN mkdir -pv $HOME/.config/atuin/
> RUN echo "host=\"0.0.0.0\"" > $HOME/.config/atuin/server.toml
> RUN echo "port=8000"              >> $HOME/.config/atuin/server.toml
> RUN echo "open_registration=true" >> $HOME/.config/atuin/server.toml
> RUN echo "db_uri=\"postgres://atuin:dbPassword@postgres.homelab.local/atuin\"" >>  $HOME/.config/atuin/server.toml
> RUN echo "#!/bin/bash" > $HOME/startServer
> RUN echo "source $HOME/.bashrc" >> $HOME/startServer
> RUN echo "source $HOME/.cargo/env" >> $HOME/startServer
> RUN echo "atuin server start" >> $HOME/startServer
> RUN chmod -R og-rwx $HOME/.config
> RUN chmod +x $HOME/startServer
> ENV ATUIN_CONFIG_DIR /root/.config/atuin
> ENV ATUIN_PORT 8000
> CMD $HOME/startServer
> ```
> 
> Even if you're not using docker, you can probably see how the atuin server is configured and started.
> 
> It's not documentation, but hope it helps.

Thanks for the input!

I'd be deploying this in kubernetes, so a container image is a must. I'd rather not stray from the official image, although `atuin server` does seem like it doesn't even support an environment variable for the postgres URI.

Would be nice to get more docs, env vars, and overall a better experience self-hosting. @ellie

**Comment by conradludgate** [comment]
I have a fix for the dockerfile, thanks for pointing that out.

As for the documentation, there is a doc in this repo: https://github.com/ellie/atuin/blob/main/docs/server.md

You can specify the Postgres DB URI either by setting `db_uri` in your `~/.config/atuin/server.toml` file, or using the `ATUIN_DB_URI` environment variable. All configurations in that toml file can be similarly configured with the matching `ATUIN_*` environment variable

**Comment by 0dragosh** [comment]
> I have a fix for the dockerfile, thanks for pointing that out.
> 
> As for the documentation, there is a doc in this repo: https://github.com/ellie/atuin/blob/main/docs/server.md
> 
> You can specify the Postgres DB URI either by setting `db_uri` in your `~/.config/atuin/server.toml` file, or using the `ATUIN_DB_URI` environment variable. All configurations in that toml file can be similarly configured with the matching `ATUIN_*` environment variable

This is very helpful, thank you!

I also couldn't find the name of the official image/repo.

**Comment by ellie** [comment]
> is lacking completely

It's linked from the README, has been for a long time, so I would hardly say it's lacking _completely_ ;)

I agree it might need filling out though. Thanks for the help there @conradludgate!

I have an image here: https://hub.docker.com/r/ellieh/atuin

Let's consider this closed with #198

**Comment by Syzygianinfern0** [comment]
Could there be a section of the docs for #277. I'm not sure how to use the custom server after getting it running.
